src/core/math.d

```diff
@@ -232,35 +235,29 @@ real rint(real x) @safe pure nothrow;      /* intrinsic */
 
 version (LDC)
 {
-    version (X86)    version = X86_Any;
-    version (X86_64) version = X86_Any;
-
-    version (X86_Any)
+    static if (isRealX87)
     {
-        static if (real.mant_dig == 64)
+        pragma(inline, true)
+        real yl2x(real x, real y)   @safe pure nothrow
         {
-            // y * log2(x)
-            real yl2x(real x, real y)   @safe pure nothrow
-            {
-                real r;
-                asm @safe pure nothrow @nogc { "fyl2x" : "=st" (r) : "st" (x), "st(1)" (y) : "st(1)"; }
-                return r;
-            }
+            real r = void;
+            asm @safe pure nothrow @nogc { "fyl2x" : "=st" (r) : "st(1)" (y), "st" (x) : "st(1)", "flags"; }
```

Comment:
Swapping the input order from `"st" (x), "st(1)" (y)` to `"st(1)" (y), "st" (x)` gets rid of an implicit `fxch` instruction. FWIW, I've also added the `~{flags}` clobber.